### DISCUSSION TITLE
Improve documentation and testing

### DISCUSSION THREAD

**Discussion by matuella** [singerdmx/flutter-quill#179] [General]
I think this library has a lot of potential of being the most useful solution to building WYSIWYG editors, but I think it lacks a lot of documentation and testing (unit). My suggestion here is to discuss about how we can address these issues and create actionable (could be a project or even a couple of issues) to improve this library.

What do you think?

**Comment by singerdmx** [comment]
Your call.
You are collaborator now.
Please take on some ownership

**Comment by friebetill** [comment]
The code base is based on [Zefyr](https://pub.dev/packages/zefyr)'s code and the code there is very well documented. A large part of the documentation can be copied almost 1 to 1. I think we also should align our license to Zefyr's in this process, because imho it's not nice to copy code and simply change the license.

**Reply by matuella** [reply]
Didn't knew about Zefyr. I completely agree with you. We could try to reach out to the zefyr owners about their BSD License and if they are fine with this new library being almost a full-fork of their project, and if they would also want to collaborate on improving this library.

**Reply by friebetill** [reply]
I am not an expert of licenses, but as far as I understand it, the BSD[-style license](https://softwareengineering.stackexchange.com/a/240620/378804) of Zefyr allows us to maintain a fork. I guess we have to mark which code has a BSD style license though. 

We can also ask the maintainer, but I guess it doesn't accomplish much, since the maintainer has not much time, unfortunately. That's why this fork came into existence, as many issues and PRs have gone unanswered for years now.
